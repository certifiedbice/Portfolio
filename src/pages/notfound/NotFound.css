#not-found-container{
position: relative;
	display: flex;
    flex-direction: column;
	align-items: center;
    justify-content: center;
	height: 100%;
	width: 100%;
}
#not-found-message{
	align-self: center;
}
#not-found-message h2{
	font-family: '3_thehard_way_rmxfenotype';
	font-size: 24px;
  	line-height: 2;
	color: #fff;
}
#back-to-home-message{
	align-self: center;
	font-family: Verdana, Geneva, Tahoma, sans-serif;
}