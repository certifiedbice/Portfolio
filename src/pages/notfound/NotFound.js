import React,{Component} from 'react';
import Context from '../../masterProvider';
import './NotFound.css';
// export default function (){
export default class NotFound extends Component{
	static contextType=Context;
	componentDidMount(){
		// Start background movement.
		this.context.bgInt(true);
		// Start page fade.
		this.context.pageFade();
		// Handle change to the home screen.
		const forwardToHome=()=>{
			this.props.history.push('/home');
		}
		const body=document.querySelector('body');
		body.addEventListener('click',function(){forwardToHome();},false);
	}
	componentWillUnmount(){
		// End background movement.
		this.context.bgInt(false);		
	}
	render(){
		return(
			<>
			<div id='background-image'></div>
			<div id='not-found-container'>
				<div id='not-found-message'>
					<h2>404 - Not Found</h2>
				</div>
				<div id='back-to-home-message'>Click or tap somewhere...</div>
			</div>
			<div id='page-fade'></div>
			</>
		);
	}
}
