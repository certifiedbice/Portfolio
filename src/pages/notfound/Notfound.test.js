import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NotFound from './NotFound';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<NotFound/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<NotFound/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});