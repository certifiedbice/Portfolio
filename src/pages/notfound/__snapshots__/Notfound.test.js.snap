// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    id="background-image"
  />,
  <div
    id="not-found-container"
  >
    <div
      id="not-found-message"
    >
      <h2>
        404 - Not Found
      </h2>
    </div>
    <div
      id="back-to-home-message"
    >
      Click or tap somewhere...
    </div>
  </div>,
  <div
    id="page-fade"
  />,
]
`;
